src/OrleansRuntime/Versions/GrainVersionStore.cs

```diff
@@ -0,0 +1,89 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Orleans.Runtime.Configuration;
+using Orleans.Storage;
+using Orleans.Versions;
+using Orleans.Versions.Compatibility;
+using Orleans.Versions.Selector;
+
+namespace Orleans.Runtime.Versions
+{
+    internal class GrainVersionStore : IVersionStore
+    {
+        private readonly IInternalGrainFactory grainFactory;
+        private readonly string deploymentId;
+        private IVersionStoreGrain StoreGrain => this.grainFactory.GetGrain<IVersionStoreGrain>(this.deploymentId);
```

Comment:
It was a cheap way to ensure that we don't use rules that were set with an older deploymentId.